ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -630,14 +700,36 @@ impl Module {
         ExtraArgsIdx::new(idx)
     }
 
+    /// Push a new type into the type table and return its index.
+    fn push_type(&mut self, ty: Type) -> Result<TypeIdx, CompilationError> {
```

Comment:
I now realise that you've effectively implemented https://crates.io/crates/indexmap. No need to fix it in this PR but might be worth considering for an upcoming PR?

Reply:
Ah yes. I vaguely remember this from our rustc days.

And it looks like it does the key parameterisation too, so perhaps all `TiVec`s should in fact be `IndexMap`s?

One for a future PR, yes.